lib/govuk_message_queue_consumer/batch_consumer.rb

```diff
@@ -0,0 +1,76 @@
+module GovukMessageQueueConsumer
+  class BatchConsumer < Consumer
+    DEFAULT_BATCH_SIZE = 100
+    DEFAULT_BATCH_TIMEOUT = 5
+    # we want to increase the prefetch size here to the batch size
+    # so that we don't need to do multiple fetches for a batch.
+    NUMBER_OF_MESSAGES_TO_PREFETCH = DEFAULT_BATCH_SIZE
+
+    def initialize(options={})
+      opts = options.dup
+      @batch_size = opts.delete(:batch_size) || DEFAULT_BATCH_SIZE
+      @batch_timeout = opts.delete(:batch_timeout) || DEFAULT_BATCH_TIMEOUT
+      super(opts)
+    end
+
+    def run
+      raise 'Already started' if @started
+      @started = true
+      @rabbitmq_connection.start
+      @running = true
+      Thread.abort_on_exception = true
+      @running_thread = Thread.new do
+        while @running do
+          process_batch
+        end
+      end
+    end
+
+    def stop(wait: true)
+      @running = false
+      @running_thread.join if wait
+    end
+
+  private
+
+    def process_batch
+      messages = []
+      with_timeout do
+        while messages.count < @batch_size do
+          delivery_info, headers, payload = queue.pop(manual_ack: true)
+          messages << Message.new(payload, headers, delivery_info) if payload
+        end
+      end
+
+      if messages.any?
+        @statsd_client.count("#{@queue_name}.started", messages.count)
+        @statsd_client.increment("#{@queue_name}.batch_started")
+        processor_chain.process(messages)
```

Comment:
Because it is a `chain` and each item on the chain calls its child process in the chain. As such if we want to process a batch to the bottom of the chain each item on the way must be able to deal with a batch. 

We could change it to a series of list processors, but that is a different thing. :)